File: examples/C/hwserver.c
//  Hello World server

#include <zmq.h>
#include <stdio.h>
#include <unistd.h>
#include <string.h>
#include <assert.h>

int main (void)
{
    //  Socket to talk to clients
    void *context = zmq_ctx_new ();
    void *responder = zmq_socket (context, ZMQ_REP);
    int response = zmq_bind (responder, "tcp://*:5555");
    assert (response == 0); // Check if the response code indicates success

    while (1) {
        static const size_t kReadBufferLength = 10;
        char buffer [kReadBufferLength];
        zmq_recv (responder, buffer, kReadBufferLength, 0);
        printf ("Received Hello\n");
        sleep (1);          //  Pretend to do some 'work'
        static const char kReplyString[] = "World";
        zmq_send(responder, kReplyString, sizeof(kReplyString) - 1, 0);
    }
    return 0;
}

File: examples/Perl/hwserver.pl
# Hello World server in Perl

use strict;
use warnings;
use v5.10;

use ZMQ::FFI;
use ZMQ::FFI::Constants qw(ZMQ_REP);

# Socket to talk to clients
my $context = ZMQ::FFI->new();
my $responder = $context->socket(ZMQ_REP);
$responder->bind("tcp://*:5555");

while (1) {
    $responder->recv();
    say "Received Hello";
    sleep 1;
    $responder->send("World");
}
